src/pino_ode_solve.jl

```diff
@@ -0,0 +1,219 @@
+"""
+	PINOODE(chain,
+	    OptimizationOptimisers.Adam(0.1),
+	    bounds;
+	    init_params = nothing,
+	    strategy = nothing
+	    kwargs...)
+
+Algorithm for solving paramentric ordinary differential equations using a physics-informed
+neural operator, which is used as a solver for a parametrized `ODEProblem`.
+
+## Positional Arguments
+
+* `chain`: A neural network architecture, defined as a `Lux.AbstractExplicitLayer` or `Flux.Chain`.
+                 `Flux.Chain` will be converted to `Lux` using `adapt(FromFluxAdaptor(false, false), chain)`
+* `opt`: The optimizer to train the neural network.
+* `bounds`: A dictionary containing the bounds for the parameters of the neural network
+in which will be train the prediction of parametric ODE.
+
+## Keyword Arguments
+
+* `init_params`: The initial parameter of the neural network. By default, this is `nothing`,
+                             which thus uses the random initialization provided by the neural network library.
+* `strategy`: The strategy for training the neural network.
+* `additional_loss`: additional function to the main one. For example, add training on data.
+* `kwargs`: Extra keyword arguments are splatted to the Optimization.jl `solve` call.
+
+## References
+
+* Sifan Wang "Learning the solution operator of parametric partial differential equations with physics-informed DeepOnets"
+* Zongyi Li "Physics-Informed Neural Operator for Learning Partial Differential Equations"
+"""
+struct PINOODE{C, O, B, I, S <: Union{Nothing, AbstractTrainingStrategy},
+    AL <: Union{Nothing, Function}, K} <:
+       SciMLBase.AbstractODEAlgorithm
+    chain::C
+    opt::O
+    bounds::B
+    init_params::I
+    strategy::S
+    additional_loss::AL
+    kwargs::K
+end
+
+function PINOODE(chain,
+        opt,
+        bounds;
+        init_params = nothing,
+        strategy = nothing,
+        additional_loss = nothing,
+        kwargs...)
+    !(chain isa Lux.AbstractExplicitLayer) && (chain = Lux.transform(chain))
+    PINOODE(chain, opt, bounds, init_params, strategy, additional_loss, kwargs)
+end
+
+struct PINOPhi{C, S}
```

Comment:
@ChrisRackauckas oh actually it should be mutable, how it is done for `ODEPhi`  

https://github.com/SciML/NeuralPDE.jl/blob/ecc6a408283c069d129b797e72a27b2817be4748/src/ode_solve.jl#L103C1-L111C4

Reply:
that should probably be made immutable too 😅

Reply:
Ah, I can look into that

Reply:
https://github.com/SciML/NeuralPDE.jl/pull/864